ecms_base/modules/custom/ecms_icon_library/src/Plugin/Field/FieldFormatter/EcmsIconLibraryFormatter.php

```diff
@@ -56,25 +56,53 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
         $file_url = file_create_url($file_uri);
 
         // Render as SVG tag.
-        $svgRaw = file_get_contents($file_url);
+        $svgRaw = $this->urlGetContents($file_url);
 
-        if ($svgRaw) {
-          $svgRaw = preg_replace(
-            ['/<\?xml.*\?>/i', '/<!DOCTYPE((.|\n|\r)*?)">/i'],
-            '',
-            $svgRaw
+        if ($svgRaw === FALSE) {
+          \Drupal::logger('ecms_icon_library')->notice(
+            'File ID ' . $fid . ' cannot be loaded as SVG.'
           );
-          $svgRaw = trim($svgRaw);
-
-          $elements[$delta]['media_library_icon'] = [
-            '#markup' => Markup::create($svgRaw),
-          ];
+          continue;
         }
+
+        $svgRaw = preg_replace(
+          ['/<\?xml.*\?>/i', '/<!DOCTYPE((.|\n|\r)*?)">/i'],
+          '',
+          $svgRaw
+        );
+        $svgRaw = trim($svgRaw);
+
+        $elements[$delta]['media_library_icon'] = [
+          '#markup' => Markup::create($svgRaw),
+        ];
       }
     }
 
     return $elements;
 
   }
 
+  /**
+   * Replacement for function file_get_contents().
+   *
+   * See stackoverflow.com/questions/3979802/alternative-to-file-get-contents.
+   *
+   * From curl_exec() docs: "If the CURLOPT_RETURNTRANSFER option is set,
+   * it will return the result on success, false on failure."
+   */
+  private function urlGetContents($url) {
```

Comment:
Add argument type and return type

Reply:
May require some logic rework based on https://www.php.net/manual/en/function.curl-exec.php
return type